patterns/grok-patterns

```diff
@@ -34,7 +34,7 @@ HOSTPORT %{IPORHOST}:%{POSINT}
 
 # paths
 PATH (?:%{UNIXPATH}|%{WINPATH})
-UNIXPATH (/([\w_%!$@:.,+~-]+|\\.)*)+
+UNIXPATH (/[\w_%!$@:.,+~-]*)+
```

Comment:
really wanted to avoid the 3-level nested loop but I am certainly no way near a regexp engine expert.

there's been a [separate attempt](https://github.com/logstash-plugins/logstash-patterns-core/pull/166) using atomic grouping. please let me know if the change seems to be "breaking" compatibility so we reconsider the alternative here.